scripts/data_processing/public/archetypes_over_time.R

```diff
@@ -0,0 +1,344 @@
+  summarise(`Insufficient data` = sum(`Insufficient data`), 
+            `Connect/Build` = sum(`Connect/Build`), 
+            `Strengthen/Leverage` = sum(`Strengthen/Leverage`), 
+            Sustain = sum(Sustain)) %>% 
+  add_column(world_bank_economies = "All", .after = "date") %>% 
+  bind_rows(Arch_seqtab)
+Arch_seqtab <- Arch_seqtab[order(match(Arch_seqtab$world_bank_economies, c("All", "High income", "Upper middle income", "Lower middle income", "Low income"))), ] %>% 
+  arrange(date)
+
+## Testing archetypes
+Arch_testtab <- Archetypes %>%
+  group_by(date, world_bank_economies, test_arch, .drop = F) %>% 
+  summarise(number = n()) %>% 
+  pivot_wider(names_from = test_arch, values_from = number)
+# Add table of all income groups
+Arch_testtab <- ungroup(Arch_testtab) %>% 
+  group_by(date) %>% 
+  summarise(`Insufficient data` = sum(`Insufficient data`), 
+            Test = sum(Test), 
+            Sustain = sum(Sustain)) %>% 
+  add_column(world_bank_economies = "All", .after = "date") %>% 
+  bind_rows(Arch_testtab)
+Arch_testtab <- Arch_testtab[order(match(Arch_testtab$world_bank_economies, c("All", "High income", "Upper middle income", "Lower middle income", "Low income"))), ] %>% 
+  arrange(date)
+
+#### Count population in each archetype
+## Sequencing archetypes
+Arch_seqtab_pop <- Archetypes %>% 
+  group_by(date, world_bank_economies, seq_arch, .drop = F) %>% 
+  summarise(population = sum(population)) %>% 
+  pivot_wider(names_from = seq_arch, values_from = population)
+# Add table of all income groups
+Arch_seqtab_pop <- ungroup(Arch_seqtab_pop) %>% 
+  group_by(date) %>% 
+  summarise(`Insufficient data` = sum(`Insufficient data`), 
+            `Connect/Build` = sum(`Connect/Build`), 
+            `Strengthen/Leverage` = sum(`Strengthen/Leverage`), 
+            Sustain = sum(Sustain)) %>% 
+  add_column(world_bank_economies = "All", .after = "date") %>% 
+  bind_rows(Arch_seqtab_pop)
+Arch_seqtab_pop <- Arch_seqtab_pop[order(match(Arch_seqtab_pop$world_bank_economies, c("All", "High income", "Upper middle income", "Lower middle income", "Low income"))), ] %>% 
+  arrange(date)
+
+## Testing archetypes
+Arch_testtab_pop <- Archetypes %>%
+  group_by(date, world_bank_economies, test_arch, .drop = F) %>% 
+  summarise(population = sum(population)) %>% 
+  pivot_wider(names_from = test_arch, values_from = population)
+# Add table of all income groups
+Arch_testtab_pop <- ungroup(Arch_testtab_pop) %>% 
+  group_by(date) %>% 
+  summarise(`Insufficient data` = sum(`Insufficient data`), 
+            Test = sum(Test), 
+            Sustain = sum(Sustain)) %>% 
+  add_column(world_bank_economies = "All", .after = "date") %>% 
+  bind_rows(Arch_testtab_pop)
+Arch_testtab_pop <- Arch_testtab_pop[order(match(Arch_testtab_pop$world_bank_economies, c("All", "High income", "Upper middle income", "Lower middle income", "Low income"))), ] %>% 
+  arrange(date)
+
+#### 9) Export ####
+#### Archetypes by N(countries)
+### NGS 
```

Comment:
I think I missed this, but is there a place where you only pull the archetypes from the first of the month? Alternatively, when you calculate the metrics for the past 12 months, instead of doing it daily, you could do it monthly on the first of the month

Reply:
In this script I do only calculate the metrics on the first of the month. I calculate everything over previous 12 month windows for each date in the object `date_seq`. See 
`### Get sequence of months to calculate sequence submissions over`
`date_seq <- seq.Date(floor_date(min(gisaid_t$submission_date, na.rm = T)+years(1), "months"), 
                     floor_date(max(gisaid_t$submission_date, na.rm = T), "months"), by = "month")
`